app/src/main/java/org/greenstand/android/TreeTracker/fragments/LoginFragment.kt

```diff
@@ -0,0 +1,200 @@
+                activateLoginButton()
+            }else {
+                v.emailEditText.addTextChangedListener(object: TextWatcher {
+                    @SuppressLint("NewApi")
+                    override fun afterTextChanged(p0: Editable?) {
+                        emailEntered = p0.toString()
+                        if (isExistingUser(emailEntered!!) || isExistingUser(phoneNumberEntered!!)) {
+                            activateLoginButton()
+                        }else
+                            when (isExistingUser(emailEntered!!)) {
+                                true -> activateLoginButton()
+                                else -> {
+                                    sign_up_button?.visibility = View.VISIBLE
+                                }
+                            }
+                    }
+
+                    override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int){
+                        inactivateLoginButton()
+                    }
+
+                    override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                        inactivateLoginButton()
+                    }
+                })
+                when (isExistingUser(phoneNumberEntered!!)) {
+                    true -> activateLoginButton()
+                    else -> {
+                        sign_up_button?.visibility = View.VISIBLE
+                    }
+                }
+
+            }
+
+        }
+
+        override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            inactivateLoginButton()
+        }
+
+        override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            inactivateLoginButton()
+        }
+
+    })
+
+    v.sign_up_button.setOnClickListener(object : View.OnClickListener{
+        override fun onClick(v: View?) {
+
+
+        }
+
+    })
+
+ return v
+}
+
+
+//This is the method that will check into DB if the textEntered(that could be the phoneNumber or the email) already
+// exists in our DB. If exists it will return TRUE else it will return FALSE. in fact, it will return the if the
+// planterId is different than -1 ( this is the initial value given to var planterId), this var will receive
+// the planterId from the DB if the user already exists
+private fun isExistingUser(textEntered: String): Boolean {
+    runBlocking {
+        planterId = GlobalScope.async {
+            TreeTrackerApplication.getAppDatabase().planterDao().getPlanterIDByIdentifier(textEntered)
+        }.await()
+
+    }
+        return planterId != -1
+
+}
+
+
+@SuppressLint("NewApi")
+fun activateLoginButton(){
+    login_button.setTextAppearance(R.style.ActiveButtonStyle)
+    login_button.setBackgroundResource(R.drawable.button_active)
+    if(sign_up_button.visibility == View.VISIBLE) sign_up_button?.visibility = View.INVISIBLE
+    login_button.setOnClickListener(object : View.OnClickListener{
```

Comment:
We can update this click listener like we did above. Anywhere else you use a click listener should just be
`myView.setOnClickListener { do something here }`